chore(pageheader): organize stories in storybook and add documentation in overview (#19520)
carbon-design-system/carbon: packages/react/src/components/PageHeader/PageHeader.mdx

    renderIcon={BeeIcon}
    contentActions={contentActions}
    pageActions={pageActions}>
    <Breadcrumb>
      <BreadcrumbItem href="/#">Breadcrumb 1</BreadcrumbItem>
      <BreadcrumbItem href="#">Breadcrumb 2</BreadcrumbItem>
    </Breadcrumb>
  </PageHeader.BreadcrumbBar>
)
```

## PageHeader.Content

<Canvas
  of={PageHeaderStories.ContentWithContextualActionsAndPageActions}
  additionalActions={[
    {
      title: 'Open in Stackblitz',
      onClick: () => stackblitzPrefillConfig(PageHeaderStories.ContentWithContextualActionsAndPageActions),
    },
  ]}
/>

The `PageHeader.Content` component defines the primary content area of the `PageHeader`, including the title, subtitle, and
any supporting text or contextual actions. It accepts a `title` prop to display the main heading and can optionally include a `renderIcon` prop
to show an icon adjacent to the title. Child components such as `PageHeader.ContentText` can be used to provide additional descriptive text.
To support use cases such as tags, `contextualActions` can be passed as a prop to render components beside the content.
`pageActions` allows integration of action buttons aligned with the content section.

> The `PageHeader.ContentPageActions` component provides responsive behavior by collapsing actions into a `MenuButto`n when the viewport

Reviewer: ```suggestion
The `PageHeader.ContentPageActions` component provides responsive behavior by collapsing actions into a `MenuButton` when the viewport
```

width is reduced. To enable this functionality, it expects an array of action objects with a specific API shape, as demonstrated below:

```jsx

const actionItems = [
  {
    // props used for both collapse menu item and non-collapsed action form
    id: 'action1',
    onClick: () => console.log(`Action 1`),
    // component to render when non-collapsed
    body: (
      <Button
        renderIcon={AiGenerate}
        iconDescription="Icon Description 1"
        hasIconOnly
        size="md"
        kind="ghost"
      />
    ),
    // props to pass to the corresponding collapsed menu item
    menuItem: {
      label: 'action 1',
    },
  },
  ...
];

return (
   <PageHeader.Content
      title="Title text"
      contextualActions={
        <>
          <Tag className="tag" type="blue" size="lg">

>             Moop

Reviewer: Just curious, what is a moop?

Author: no idea 😂 I'll rename it "Tag"

  ]}
/>

## PageHeader.BreadcrumbBar

The `PageHeader.BreadcrumbBar` component is used to render the breadcrumb navigation area within the `PageHeader`.

> It accepts `Breadcrumb` and `BreadcrumbItem` components as children to define the breadcrumb trail.Additionally, it accepts

Reviewer 2: ```suggestion
It accepts `Breadcrumb` and `BreadcrumbItem` components as children to define the breadcrumb trail. Additionally, it accepts
```

`contentActions` and` pageActions` props, allowing for actions, such as `Button` or `IconButton` — alongside
the breadcrumb content.

```jsx

> import { Bee } from '@carbon/icons-react';

Reviewer 2: ```suggestion
import { Bee, Activity } from '@carbon/icons-react';
```
